introspection-engine/connectors/sql-introspection-connector/src/defaults.rs

```diff
@@ -1,126 +1,153 @@
+            Some(renderer::DefaultValue::function(Function::new("autoincrement")))
+        }
+        (Some(sql::DefaultKind::Now), sql::ColumnTypeFamily::DateTime) => {
+            Some(renderer::DefaultValue::function(Function::new("now")))
+        }
+        (Some(sql::DefaultKind::DbGenerated(default_string)), _) => {
+            let mut fun = Function::new("dbgenerated");
+
+            if let Some(param) = default_string.as_ref().filter(|s| !s.trim_matches('\0').is_empty()) {
+                fun.push_param(Value::from(Text::new(param)));
+            }
+
+            Some(renderer::DefaultValue::function(fun))
         }
-        (_, sql::ColumnTypeFamily::Int) if column.is_autoincrement() => Some(dml::DefaultValue::new_expression(
-            dml::ValueGenerator::new_autoincrement(),
-        )),
-        (_, sql::ColumnTypeFamily::BigInt) if column.is_autoincrement() => Some(dml::DefaultValue::new_expression(
-            dml::ValueGenerator::new_autoincrement(),
-        )),
-        (_, sql::ColumnTypeFamily::Int) if is_sequence(column) => Some(dml::DefaultValue::new_expression(
-            dml::ValueGenerator::new_autoincrement(),
-        )),
-        (_, sql::ColumnTypeFamily::BigInt) if is_sequence(column) => Some(dml::DefaultValue::new_expression(
-            dml::ValueGenerator::new_autoincrement(),
-        )),
-        (Some(sql::DefaultKind::Sequence(_)), _) => Some(dml::DefaultValue::new_expression(
-            dml::ValueGenerator::new_autoincrement(),
-        )),
-        (Some(sql::DefaultKind::Now), sql::ColumnTypeFamily::DateTime) => Some(set_default(
-            dml::DefaultValue::new_expression(dml::ValueGenerator::new_now()),
-            column,
-        )),
-        (Some(sql::DefaultKind::DbGenerated(default_string)), _) => Some(set_default(
-            dml::DefaultValue::new_expression(dml::ValueGenerator::new_dbgenerated(
-                default_string.as_ref().unwrap().clone(),
-            )),
-            column,
-        )),
         (Some(sql::DefaultKind::Value(dml::PrismaValue::Enum(variant))), sql::ColumnTypeFamily::Enum(enum_id)) => {
             let variant = ctx
                 .schema
                 .walk(*enum_id)
                 .variants()
                 .find(|v| v.name() == variant)
                 .unwrap();
+
             let variant_name = ctx.enum_variant_name(variant.id).prisma_name();
-            Some(set_default(
-                dml::DefaultValue::new_single(dml::PrismaValue::Enum(variant_name.into_owned())),
-                column,
-            ))
+            Some(renderer::DefaultValue::constant(variant_name))
         }
-        (Some(sql::DefaultKind::Value(val)), _) => {
-            Some(set_default(dml::DefaultValue::new_single(val.clone()), column))
+        (Some(sql::DefaultKind::Value(dml::PrismaValue::String(val))), _) => Some(renderer::DefaultValue::text(val)),
+        (Some(sql::DefaultKind::Value(dml::PrismaValue::List(val))), _) => {
```

Comment:
I'd be happy to get rid of `PrismaValue` in intro. The `Display` there doesn't work very well for us, and changing it might break something in the upstream, which I'm not risking now.

Reply:
yes we should think about a replacement for PrismaValue in sql-schema-describer